getnacelle/nacelle-react
# [gatsby-source-nacelle]: move remote image fetching to sourceNodes

**Story:** [LS-1436]([#url-here](https://nacelle.atlassian.net/browse/LS-1436)) 

## Type of Change


## What is being changed and why?

Moves fetching of remote image files to `sourceNodes` in `gatsby-source-nacelle` before the related nodes.  This PR fixes the bug with Gatsby 4 that prevents remote images from being attached to `NacelleMedia` nodes at build time. This bug existed because Gatsby 4 stops tracking node changes outside of specific actions. 

Because we don't lean on any Gatsby 4 specific lifecycles/methods, this should be a non-breaking change for consumers.




## How to Test

### Test `example/gatsby`

1. Clone this branch (`LS-1436-fix-gatsby-remote-image-in-gatsby-4`) locally. 
2. Run `npm run bootstrap` at the root of the repo
3. Navigate to `examples/gatsby` 
4. Run `npm run dev` and verify things work
5. Run `npm run build` and verify the build is successful

### Test `gatsby-source-nacelle-x-gatsby-image`

1. Navigate to `packages/gatsby-plugin-image` in your terminal and run `npm link`. 
2. Clone [getnacelle/gatsby-source-nacelle-x-gatsby-image](https://github.com/getnacelle/gatsby-source-nacelle-x-gatsby-image) & run `npm install` in that repo.
3. Run `npm link @nacelle/gatsby-source-nacelle`
4. Run `npm run build`
5. You should see `~Gatsby Image Data ~` with an object underneath it in your terminal
   - Note if you see `undefined` then this PR is not working correctly
